Client/Frontend/ContentBlocker/ContentBlockerHelper.swift

```diff
@@ -17,7 +17,37 @@ class ContentBlockerHelper {
     fileprivate weak var tab: Tab?
     fileprivate weak var profile: Profile?
 
-    static var blockImagesRule: WKContentRuleList?
+    static fileprivate var blockImagesRule: WKContentRuleList?
+    static fileprivate var whitelistedDomains = [String]()
+
+    // Only set and used in UI test
+    static weak var testInstance: ContentBlockerHelper?
+
+    func whitelist(enable: Bool, forDomain domain: String, completion: (() -> Void)?) {
+        if enable {
+            ContentBlockerHelper.whitelistedDomains.append(domain)
+        } else {
+            ContentBlockerHelper.whitelistedDomains = ContentBlockerHelper.whitelistedDomains.filter { $0 != domain }
+        }
+
+        removeAllRulesInStore {
+            self.compileListsNotInStore {
+                NotificationCenter.default.post(name: .ContentBlockerUpdateNeeded, object: nil)
+                completion?()
+            }
+        }
```

Comment:
When the rules change, the lists need to be removed, re-added, and the notification sent for all tabs to re-install the active blocking rules.